### ISSUE TITLE
Error sourceUrl from `securestorage.js`

### ISSUE THREAD

**Issue by karlbateman** [Crypho/cordova-plugin-secure-storage#87]
Hey all,
I am trying to get the secure storage module intagrated as service.

I have the following so far, to test all is working..

```typescript
import { Injectable } from '@angular/core';
import { Platform } from 'ionic-angular';
import { SecureStorage } from 'ionic-native';
import 'rxjs/add/operator/map';

@Injectable()
export class UserService {

  constructor(private platform: Platform) {
    platform.ready().then(() => {
      let storage: SecureStorage = new SecureStorage();

      storage.create('aspect-user')
      .then(
        () => console.log('SecureStore created'),
        error => console.log(`Error, SecureStore could not be created ${JSON.stringify(error)}`)
      );

      storage.set('user', 'some_user')
      .then(
        () => console.log('User created'),
        error => console.log(`Error, user could not be created ${JSON.stringify(error)}`)
      )
    });
  }
}
```

Running the above with `ionic emulate ios -lc`, gives the following output;

```bash
console.log: Error, user could not be created {"line":71,"column":45,"sourceURL":"http://192.168.1.175:8100/plugins/cordova-plugin-secure-storage/www/securestorage.js"}
```

What am I missing??

**Comment by ggozad** [comment]
Will this perhaps be better addressed in some ionic forum?

**Comment by greigdendor** [comment]
Appears to be related to https://github.com/Crypho/cordova-plugin-secure-storage/issues/48

**Comment by demetris-manikas** [comment]
@greigdendor 
Sorry for the late reply but there is a common error in your code. You try to use the storage without waiting for its creation to finish. Move the set inside the success callback of create and it will work as expected.

**Comment by demetris-manikas** [comment]
@ggozad  the examples of set and get should be revised to include the create method.

**Comment by karlbateman** [comment]
How would `set`/`get` be used outside of the `create` method? Would this be best called during `initializeApp()`?

**Comment by demetris-manikas** [comment]
As documentation states you only have to make sure that the success callback of the create has fired before calling any other method on the storage.
There are several ways to achieve this but here is not the place to explain them. 
Look around and pick the one that suits you best.

**Comment by greigdendor** [comment]
@demetris-manikas 
No problem about late reply. You are right of course. I realised this and modified our test code, then I hit the problem with the iOS 10 emulator, hence I referred to #48 . In my haste I didn't explain how I got to that point in this issue. In any case the initial issue was resolved. Note that @karlbateman raised this issue but we work together.

@karlbateman 
In reality the storage object would be created in the constructor, and get/set would be methods which are used after the storage object had been created (and existence checked). If you always wanted to create a user when instantiating the object, you'd pass the user info into the constructor and, as @demetris-manikas says, you would put set in the callback of create.

This is now our own internal implementation decision so this issue can now be closed, many thanks for all input.
